core/blockchain_test.go

```diff
@@ -471,11 +471,14 @@ func makeBlockChainWithDiff(genesis *types.Block, d []int, seed byte) []*types.B
 
 func chm(genesis *types.Block, db ethdb.Database) *BlockChain {
 	var eventMux event.TypeMux
-	bc := &BlockChain{chainDb: db, genesisBlock: genesis, eventMux: &eventMux, pow: FakePow{}, rand: rand.New(rand.NewSource(0))}
-	bc.headerCache, _ = lru.New(100)
+	bc := &BlockChain{chainDb: db, genesisBlock: genesis, eventMux: &eventMux, pow: FakePow{}}
+	bc.hc = &HeaderChain{chainDb: db, genesisHeader: genesis.Header(), rand: rand.New(rand.NewSource(0))}
+	bc.hc.getValidator = func() HeaderValidator { return bc.Validator() }
+	bc.hc.procInterrupt = bc.getProcInterrupt
```

Comment:
I'm using NewHeaderChain directly now, but I can't insert it into bc init because for hc init we need a validator function which references bc.